src/main/java/com/puppycrawl/tools/checkstyle/checks/annotation/AnnotationLocationCheck.java

```diff
@@ -264,21 +264,11 @@ public final void setAllowSamelineMultipleAnnotations(boolean allow) {
     public void visitToken(DetailAST ast) {
         final DetailAST modifiersNode = ast.findFirstToken(TokenTypes.MODIFIERS);
 
-        if (hasAnnotations(modifiersNode)) {
+        if (modifiersNode != null) {
```

Comment:
Inside `hasAnnotations` the check for having an annotation is redundant as `checkAnnotations` already has another check for only working on annotations.